src/main/java/com/hackathonorganizer/hackathonwriteservice/hackathon/service/HackathonService.java

```diff
@@ -0,0 +1,92 @@
+package com.hackathonorganizer.hackathonwriteservice.hackathon.service;
+
+import com.hackathonorganizer.hackathonwriteservice.hackathon.exception.ResourceNotFoundException;
+import com.hackathonorganizer.hackathonwriteservice.hackathon.model.Hackathon;
+import com.hackathonorganizer.hackathonwriteservice.hackathon.model.HackathonDto;
+import com.hackathonorganizer.hackathonwriteservice.hackathon.repository.HackathonRepository;
+import com.hackathonorganizer.hackathonwriteservice.hackathon.utils.HackathonMapper;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+@Slf4j
+public class HackathonService {
+
+    private final HackathonRepository hackathonRepository;
+
+    public HackathonDto createHackathon(Hackathon hackathon) {
+
+        Hackathon savedHackathon = hackathonRepository.save(hackathon);
+
+        log.info("Hackathon with id: " + savedHackathon.getId() + " saved successfully");
+
+        return HackathonMapper.mapHackathonToDto(savedHackathon);
+    }
+
+    public HackathonDto updateHackathonData(Hackathon hackathonUpdatedData) {
+
+        Hackathon hackathon = hackathonRepository.findById(hackathonUpdatedData.getId())
+                .orElseThrow(() ->
+                        new ResourceNotFoundException(String.format("Hackathon with id: %d not found", hackathonUpdatedData.getId())));
+
+        Hackathon.HackathonBuilder hackathonBuilder = hackathon.toBuilder();
+
+        Hackathon updatedHackathon = hackathonBuilder
+                .name(hackathonUpdatedData.getName())
+                .description(hackathonUpdatedData.getDescription())
+                .organizerInfo(hackathonUpdatedData.getOrganizerInfo())
+                .eventStartDate(hackathonUpdatedData.getEventStartDate())
+                .eventEndDate(hackathonUpdatedData.getEventEndDate())
+                .build();
+
+
+        Hackathon savedHackathon = hackathonRepository.save(updatedHackathon);
+
+        log.info("Hackathon with id: " + savedHackathon.getId() + " updated successfully");
+
+        return HackathonMapper.mapHackathonToDto(savedHackathon);
+    }
+
+    public String deactivateHackathon(Long hackathonId) {
+
+        Hackathon hackathon = hackathonRepository.findById(hackathonId)
+                .orElseThrow(() -> new ResourceNotFoundException(String.format("Hackathon with id: %d not found", hackathonId)));
+
+        hackathon.setActive(false);
+
+        hackathonRepository.save(hackathon);
+
+        log.info("Hackathon with id: " + hackathonId + " deactivated successfully");
```

Comment:
Jak juz cos logujemy, proponuje skorzystac z aspectow i tam logowac poszczegolne informacje

Reply:
aspektami raczej ogarniałbym tylko bardziej generyczne logi np. timingi metod czy jakieś monitorowanie aktywności i statystyki jeśli już. takie typowo serwisowe logi wyrzucać do aspektów to tylko komplikowanie sobie życia